content/en/tracing/guide/instrument_custom_method.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Instrument a custom method to get deep visibility into your business logic
+kind: guide
+disable_toc: true
+further_reading:
+- link: "/tracing/guide/alert_anomalies_p99_database/"
+  tag: "3 mins"
+  text: "Alert on anomalous p99 latency of a database service"
+- link: "tracing/guide/week_over_week_p50_comparison/"
+  tag: "2 mins"
+  text: "Compare a service’s latency to the previous week"
+- link: "/tracing/guide/slowest_request_daily/"
+  tag: "3 mins"
+  text: "Debug the slowest trace on the slowest endpoint of a web service"
+- link: "tracing/guide/"
+  tag: ""
+  text: "All guides"
+---
+_8 minutes to complete_
+
+{{< img src="tracing/guide/custom_span/custom_span_1.png" alt="Analytics View" responsive="true" style="width:90%;">}}
+
+In order to provide you deep visibility into your business logic, Datadog APM allows you to customize the spans that make up your traces based on your needs and implementation. This empowers you to trace any method in your codebase and even specific components within methods. You can use this to optimize and monitor critical areas of your application at the granularity that works for you.
+
+Datadog instruments many frameworks out-of-the-box such as web services, databases and caches and it enables you to instrument your own business logic to have the exact visibility you need. By creating spans for the methods, you can optimize timing and track errors using the APM flamegraph and Datadog monitors.
+
+## Instrument a complete method with decorators
```

Comment:
For languages without decorator support, this part of the document is not very clear to follow.

Given the next section, `Instrument individual components of your methods for better visibility into their performance`, reads "this example walks through adding child spans to the `BackupLedger.write` span created above.", we need to provide a "decorator" example for all languages.

In my case, Ruby does not support decorators, so I provided an example on how to instrument the `BackupLedger.write` as a whole in this section. With that, the next section reads correctly.

Maybe we could reword this section to be `Instrument a complete method` instead, without assuming the presence of decorators, and only mention decorators on languages that support it.

Reply:
I completely redid these, wondering what you think of the new version?